File: analysis/import/SCMLogParser/src/main/java/de/maibornwolff/codecharta/model/input/metrics/NumberOfAuthors.java
package de.maibornwolff.codecharta.model.input.metrics;

import de.maibornwolff.codecharta.model.input.Commit;

import java.util.HashSet;
import java.util.Set;

public final class NumberOfAuthors implements CommitMetric<Integer> {
    private final Set<String> authors = new HashSet<>();

    @Override
    public String metricName() {
        return "number_of_authors";
    }

    @Override
    public void registerCommit(Commit commit) {
        authors.add(commit.getAuthor());
    }

    @Override
    public Integer value() {
        return authors.size();
    }
}

File: analysis/import/SCMLogParserV2/src/main/kotlin/de/maibornwolff/codecharta/importer/scmlogparserv2/input/metrics/NumberOfAuthors.kt
package de.maibornwolff.codecharta.importer.scmlogparserv2.input.metrics

import de.maibornwolff.codecharta.importer.scmlogparserv2.input.Commit
import de.maibornwolff.codecharta.model.AttributeType

class NumberOfAuthors : Metric {
    private val authors = mutableSetOf<String>()

    override fun description(): String {
        return "Number of Authors: Number of authors that contributed to this file."
    }

    override fun metricName(): String {
        return "number_of_authors"
    }

    override fun registerCommit(commit: Commit) {
        authors.add(commit.author)
    }

    override fun value(): Number {
        return authors.size
    }

    override fun attributeType(): AttributeType {
        return AttributeType.relative
    }
}
